timesheet_activity_report/report/timesheet_activity_report.py

```diff
@@ -0,0 +1,115 @@
+# © 2016-2019 Elico Corp (https://www.elico-corp.com)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import models, fields, tools
+
+
+class TimesheetReport(models.Model):
+    """
+        Timesheet Activities Report.
+    """
+    _name = "timesheet.activity.report"
+    # Database table should not be created, use init() instead
+    _auto = False
+    _description = "Timesheet Activities Report"
+    # Field used for the Name
+    _rec_name = 'activity_name'
+
+    id = fields.Integer('Id', readonly=True)
+    activity_type = fields.Selection(
+        [
+            ('task', 'Task'),
+            ('issue', 'Issue'),
+            ('timesheet', 'Timesheet'),
+        ], 'Type', readonly=True,
+        help="Type is based on the origin of the input (Task, Issue or "
+             "Timesheet Activities)")
+    description = fields.Char('Description', readonly=True)
+    hours = fields.Float(
+        'Time spent', digits=(16, 2), readonly=True,
+        help="Time spent on timesheet")
+    user_id = fields.Many2one('res.users', 'User', readonly=True)
+    product_id = fields.Many2one('product.product', 'Product', readonly=True)
+    date = fields.Date('Date', readonly=True)
+    project_id = fields.Many2one('project.project', 'Project', readonly=True)
+    project_state = fields.Char('State', readonly=True, help="Project State")
+    activity_stage_id = fields.Many2one(
+        'project.task.type', 'Stage',
+        readonly=True, help="Activity Stage")
+    account_id = fields.Many2one(
+        'account.analytic.account', 'Analytic account', readonly=True)
+    activity_id = fields.Char(
+        'Activity id', readonly=True, help="Task id or Issue id")
+    activity_name = fields.Char(
+        'Activity name', readonly=True, help="Task name or Issue name")
+    categ_id = fields.Many2one(
+        'project.category.main', 'Task Cat.',
+        readonly=True, help="Task Category")
+    br_id = fields.Many2one(
+        'business.requirement', 'Bus. requ.',
+        readonly=True, help="Business requirement")
+    partner_id = fields.Many2one(
+        'res.partner', 'Customer', readonly=True)
+    project_categ_id = fields.Many2one(
+        'project.project.category',
+        'Project Cat.', readonly=True, help="Project Category")
+
+    def init(self, cr):
+        """
+            Timesheet Activities Report.
+        """
+        tools.drop_view_if_exists(cr, 'timesheet_activity_report')
+        cr.execute("""

```

Comment:
@seb-elico probably requires that you have a deep look at current model structure